app/controllers/introduce_yourself_controller.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+class IntroduceYourselfController < StepsController
+  def edit
+    current_app = current_or_new_snap_application
+
+    @step = step_class.new(
+      HashWithIndifferentAccess.
+        new(current_app.attributes).
+        slice(*step_attrs),
+    )
```

Comment:
what do you think about extracting `HashWithIndifferentAccess.new(current_app.attributes).slice(*step_attrs)` out to something like?

```ruby
def existing_attributes  # or, like below, `snap_application_attributes`
  HashWithIndifferentAccess.
    new(current_app.attributes).
    slice(*step_attrs)
end
```